sprints/sprint_3/design_use_cases.md

```diff
@@ -0,0 +1,140 @@
+# Design Use Cases
+
+## D6 (v1)
+
+1. **Title**: Logger
+2. **Purpose**: To log all simulation state changes as events and allow other processes to subscribe to these events and perform tasks such as metrics aggregation, event forwarding to external systems, etc.
+3. **Actors**: Simulation Engine
+4. **Preconditions**:
+    - The main simulation loop is running
+5. **Main Flow**:
+    - a subroutine within a simulation calls the `log` method to send an event of a specific type with an optional payload to the logger
+    - the logger broadcasts the event to all currently attached subscriber processes (processes that are running in their own green thread (go-routine) and thus external to the simulation)
+6. **Alternative Flow**:
+    - an external process attaches a subscriber to the logger, supplying a closured callback function to be called when an event is received by the logger
+    - the logger calls all attached callback functions with events as they arrive
+7. **Postcondition**:
+    - There are no side-effects (as far as the simulation is concerned) from either of these flows
+
+## D7 (v1)
+
+1. **Title**: Init Receiver (InitRx)
```

Comment:
Would this be better named something along the lines of Setup Messaging? The name suggests it is only related to InitRx rather than the rest of the messaging stack which are included in the flow